c/src/point_one/polaris/polaris.c

```diff
@@ -490,21 +490,41 @@ int Polaris_Work(PolarisContext_t* context) {
       recv(context->socket, context->recv_buffer, POLARIS_RECV_BUFFER_SIZE, 0);
 #endif
 
+#ifdef P1_FREERTOS
+  // FreeRTOS does not use errno for recv() errors, but instead returns the
+  // equivalent error value. On a socket timeout, it returns 0 instead of
+  // ETIMEDOUT.
+  //
+  // We try to mimick the actual POSIX behavior to simplify the code below.
+  if (bytes_read == 0) {
+    bytes_read = -1;
+    errno = ETIMEDOUT;
```

Comment:
Is errno defined in the freeertos context?

Reply:
Yes, I believe so. We may need to include `#include "FreeRTOS_POSIX/errno.h"` actually - will need to check quickly

Reply:
It appears to compile. It's bringing in `errno.h` from the GCC toolchain (`arm-none-eabi/include/sys/errno.h`). The FreeRTOS definitions from `projdefs.h` (e.g., `pdFREERTOS_ERRNO_ETIMEDOUT`) are intentionally defined to match the POSIX values, and all the relevant values appear to match `errno.h` as expected.